### ISSUE TITLE
Cannot get past the login screen no matter the password

### ISSUE THREAD

**Issue by likyng** [owncloud/core#24319]
### Steps to reproduce
1. Install ownCloud 9.0.1
2. Upgrade contacts app to 1.2

### Expected behaviour
I should be able to login and use owncloud.

### Actual behaviour
I can see the login screen and enter my password. However, I am not able to login - upon pressing enter the page loads and returns me back to the login page. It does not matter if I enter a valid or an invalid password. I will not receive any error message back, no matther if the password was valid. 

### Server configuration
**Operating system**:
Arch Linux

**Web server:**
nginx 1.10

**Database:**
mariadb 10.1.13

**PHP version:**
php 7.0.6

**ownCloud version:** 9.0.1.3

**Updated from an older ownCloud or fresh install:** Upgraded from 8.4 to 9.0.1

**Where did you install ownCloud from:** Package repository

**Signing status (ownCloud 9.0 and above):**

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
Enabled:
  - activity: 2.2.1
  - calendar: 1.1
  - comments: 0.2
  - contacts: 1.2.0.0
  - dav: 0.1.6
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_external: 0.5.2
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - firstrunwizard: 1.1
  - galleryplus: 15.0.0
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "xx",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "xx.xx.xx"
        ],
        "datadirectory": "\/usr\/share\/webapps\/owncloud\/data",
        "overwrite.cli.url": "https:\/\/xx.xx.xx",
        "dbtype": "mysql",
        "version": "9.0.1.3",
        "dbname": "ownclouddb",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "mail_from_address": "xx",
        "mail_smtpmode": "php",
        "mail_smtpauth": 1,
        "mail_domain": "xx.xx.xx",
        "preview_libreoffice_path": "\/usr\/bin\/libreoffice",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "asset-pipeline.enabled": false,
        "theme": "",
        "loglevel": 2,
        "log_rotate_size": "104857600",
        "maintenance": false,
        "trashbin_retention_obligation": "auto"
    }
}

```

**Are you using external storage, if yes which one:** Dropbox, webdav. disabling the external storage app does not change anything though.

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:**
Firefox 45
Safari iOS 9.3.1

**Operating system:**
Arch Linux
iOS 9.3.1

#### Web server error log
```
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/oc-backbone.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 252 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/placeholder.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 2288 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/jquery.avatar.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 4310 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/shareconfigmodel.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 2277 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/shareitemmodel.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 20285 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/sharedialogresharerinfoview.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 2610 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/sharedialoglinkshareview.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 11103 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/sharedialogexpirationview.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 5902 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/sharedialogshareelistview.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 10781 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/sharedialogview.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 12234 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/share.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 14545 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /apps/files_pdfviewer/js/previewplugin.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 4127 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /apps/firstrunwizard/js/jquery.colorbox.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 26018 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /apps/firstrunwizard/js/firstrunwizard.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 547 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/files/fileinfo.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 2036 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/files/client.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 18124 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/visitortimezone.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 385 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/vendor/jsTimezoneDetect/jstz.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 14565 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/lostpassword.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 4085 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/js/login.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 557 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/vendor/backbone/backbone.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 71415 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/img/loading-dark.gif HTTP/1.1" 200 3208 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/img/logo-icon.svg HTTP/1.1" 200 2910 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/img/actions/confirm.svg HTTP/1.1" 200 370 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/fonts/OpenSans-Light.woff HTTP/1.1" 200 62844 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/fonts/OpenSans-Regular.woff HTTP/1.1" 200 20544 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/fonts/OpenSans-Semibold.woff HTTP/1.1" 200 69888 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:56 +0200] "GET /core/img/breadcrumb.svg HTTP/1.1" 200 455 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:59 +0200] "POST /index.php/ HTTP/1.1" 200 2239 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
185.x.x.x - - [28/Apr/2016:08:37:59 +0200] "GET /index.php/core/js/oc.js?v=6e88b67840609f3719fccb3c533d8ffb HTTP/1.1" 200 2233 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0"
```

#### ownCloud log (data/owncloud.log)
```
...
...
{"reqId":"ZYAL\/MjJx30MEKyKcF1I","remoteAddr":"81.xxx.x.x.","app":"core","message":"Trusted domain error. \"81.xx.xx.x\" tried to access using \"178x.x.x\" as host.","level":2,"time":"2016-04-28T04:59:03+00:00","method":"GET","url":"\/","user":"--"}
{"reqId":"Ohm\/a+QBPElrvF9SZzSz","remoteAddr":"","app":"core","message":"starting upgrade from 9.0.1.3 to 9.0.1.3","level":0,"time":"2016-04-28T05:51:01+00:00","method":"--","url":"--","user":"--"}
```

#### Browser log
```
Insert your browser log here, this could for example include:

Content Security Policy: Directive 'frame-src' has been deprecated. Please use directive 'child-src' instead. <unknown>
SyntaxError: expected expression, got '<' oc.js:1:0
Using //@ to indicate sourceMappingURL pragmas is deprecated. Use //# instead jquery.min.js:1:0
Content Security Policy: The page's settings blocked the loading of a resource at self ("script-src https://xx.xx.xx 'unsafe-eval'"). index.php
ReferenceError: oc_appconfig is not defined
<anonymous>
 shareconfigmodel.js:24
<anonymous>
 shareconfigmodel.js:13
 shareconfigmodel.js:24:4
```

How could I proceed / help to debug the issue further? Thanks!

**Comment by PVince81** [comment]
> Trusted domain error. \"81.xx.xx.x\" tried to access using \"178x.x.x\" as host

Please fix your trusted domain in config.php and/or try connecting to the IP of the trusted domain.

Also try clearing your browser's cookies, just in case.

**Comment by likyng** [comment]
Thanks for your suggestion, however that error was not related, as the last log entry I was able to make out was the upgrade notice. Before that owncloud worked fine. I figured out what the error was however, I also upgraded PHP to 7.0.6 and it seems that 7.0.6 breaks owncloud. Fortunately, this patch: https://github.com/owncloud/core/commit/238dbe9252e50ce38449045a8c2e8115a082bd1a.patch made it work again :)! 
(Just for future reference)
